hazelcast/include/hazelcast/util/Sync.h

```diff
@@ -112,8 +112,7 @@ namespace hazelcast {
             }
 
         protected:
-
-            std::mutex mutex;
+            mutable std::mutex mutex;
```

Comment:
this change seems to conflict with what the PR says it wants to achieve. We did hide such non-const changes in the past but surely it was not a good practice since it actually really is not a const operation to acquire a lock and may be it is better to let it seen this way from outside as well, so that the method signature is more reliable.

Reply:
ok, per https://herbsutter.com/2013/01/01/video-you-dont-know-const-and-mutable/